src/simpleq/queues.py

```diff
@@ -0,0 +1,219 @@
+        return attributes.get("ApproximateNumberOfMessages")
+
+    def delete(self):
+        """
+        Delete this SQS queue.
+
+        This will remove all jobs in the queue, regardless of whether or not
+        they're currently being processed.  This data cannot be recovered.
+        """
+        if self._queue:
+            self._queue.delete()
+
+    def add_job(self, job, delay=None):
+        """
+        Add a new job to the queue.
+
+        This will serialize the desired code, and dump it into this SQS queue
+        to be processed.
+
+        :param obj job: The Job to enqueue.
+        :param int delay: Delay sending job to SQS queue.
+
+        """
+
+        if delay and job.id is not None:
+            if job.id not in self._delayed_jobs:
+                th.Timer(delay, self.add_job, args=[job]).start()
+
+            self._delayed_jobs[job.id].append(job)
+            return
+
+        if job.id in self._delayed_jobs:
+            del self._delayed_jobs[job.id]
+
+        self.queue.send_message(**job.message)
+
+    def remove_job(self, job):
+        """
+        Remove a job from the queue.
+
+        :param obj job: The Job to dequeue.
+        """
+        if job._sqs_message_id is None or job._sqs_receipt_handle is None:
+            print(f"Unable to remove job from sqs queue [{job}]")
+            return
+
+        self.queue.delete_messages(
+            Entries=[
+                dict(
+                    Id=job._sqs_message_id,
+                    ReceiptHandle=job._sqs_receipt_handle,
+                )
+            ]
+        )
+
+    def _cleanup_duplicate_jobs(self, duplicate_job_groups):
+        for duplicate_jobs in duplicate_job_groups.values():
+            for duplicate_job in duplicate_jobs:
+                self.remove_job(duplicate_job)
+
+    @property
+    def jobs(self):
+        """
+        Iterate through all existing jobs in the cheapest and quickest possible
+        way.
+
+        By default we will:
+
+            - Use the maximum batch size to reduce calls to SQS.  This will
+              reduce the cost of running the service, as less requests equals
+              less dollars.
+
+            - Wait for as long as possible (*20 seconds*) for a message to be
+              sent to us (*if none are in the queue already*).  This way, we'll
+              reduce our total request count, and spend less dollars.
+
+        .. note::
+            This method is a generator which will continue to return results
+            until this SQS queue is emptied.
+        """
+
+        messages = self.queue.receive_messages(
+            MaxNumberOfMessages=self.BATCH_SIZE,
+            WaitTimeSeconds=self.WAIT_SECONDS,
+            MessageAttributeNames=["id"],
+        )
+        duplicate_job_groups = dict()
+
+        for message in messages:
+            job = Job.from_message(message)
+            if job.id is not None and job.id in duplicate_job_groups:
```

Comment:
As I understand it, based on `content_deduplication` this should never happen, but is good to have in case `content_deduplication` was not true?